api/src/Util/GetCampFromContentNodeTrait.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace App\Util;
+
+use App\Entity\Activity;
+use App\Entity\BelongsToCampInterface;
+use App\Entity\BelongsToContentNodeTreeInterface;
+use App\Entity\Camp;
+use App\Entity\Category;
+use Doctrine\ORM\EntityManagerInterface;
+
+trait GetCampFromContentNodeTrait {
+    private function getCampFromInterface(mixed $subject, EntityManagerInterface $em): null|Camp {
+        if ($subject instanceof BelongsToCampInterface) {
+            return $subject->getCamp();
+        }
+
+        if ($subject instanceof BelongsToContentNodeTreeInterface) {
+            return $this->getCampFromContentNode($subject, $em);
+        }
+
+        return null;
+    }
+
+    private function getCampFromContentNode(BelongsToContentNodeTreeInterface $subject, EntityManagerInterface $em): null|Camp {
```

Comment:
Is there any reason why you did not make `BelongsToContentNodeTreeInterface` extend `BelongsToCampInterface` and implement the traversal in the entities (or in an Entity trait), similar to how we had it before? Traversing entity relations is traditionally done on the entity classes themselves in the PHP world, do you feel this here is a different case? I feel like traversing the relations is not really the job of the voters and validators, and having to use `mixed` / `instanceof A || instanceof B` when we really know what classes we want to handle seems like a code smell to me.

Reply:
This is one of the core changes we discussed in the last meeting https://github.com/ecamp/ecamp3/issues/2569#issuecomment-1110226313

This PR removes the owner property from the `ContentNode`, which was one of the main drivers for the slow performance of the `ContentNode` endpoint.

After removing the owner property, the `ContentNode` entity has not possibility to know it's owner. For this a manual DB query is needed, utilizing the entity manager. However, the entity itself cannot and should not know the entity manager.

The solution we came up in last meeting's discussion was to introduce a new interface `BelongsToContentNodeTreeInterface` and then give the task of resolving ownership to the Voter.

An alternative we discussed in the meeting, but rejected, was to store the camp property redundantly in each `ContentNode` (or each root `ContentNode`).